### ISSUE TITLE
Warnings in log after setting `listener.tcp.default.proxy_protocol = on`

### ISSUE THREAD

**Issue by davionxiang** [erlio/vernemq#521]
### Environment

- VerneMQ Version: 1.0.1
- OS: CentOS 6.4
- Erlang/OTP version (if building from source): 19

### Actual behaviour

I set listener.tcp.default.proxy_protocol = on and I did get the real IP of the request over ELB.
But after I changed the settings, I saw some warnings appeared in log like
```
18:01:36.665 [warn]  Transport option {:ip, {0, 0, 0, 0}} unknown or invalid.
18:01:36.665 [warn]  Transport option {:port, 1883} unknown or invalid.
```
And if I change the setting to listener.tcp.proxy_protocol = on, there is no warning, but the proxy protocol does not work.

**Comment by ioolkos** [comment]
Thanks @davionxiang for reporting.
The second point you mention is a known bug:  https://github.com/erlio/vernemq/issues/516 (proxy protocol only works when the listeners are specified literally).

I'm not sure about the first point currently. Does 127.0.0.1 IP work, or is the format the problem? have you found a workaround?

**Comment by larshesel** [comment]
I suspect we somehow pass those tuples to ranch when starting the socket as it seems this is a ranch-warning. We probably forgot to sanitize the proplist we pass on or something.

**Comment by davionxiang** [comment]
@ioolkos it works well, just always output warning to log, same warning if I change IP to 127.0.0.1

**Comment by larshesel** [comment]
The warning is harmless - I looked into it and it does indeed come from ranch when it is filtering out the allowed set of parameters passed to ranch when a socket is created. We just need to filter out those parameters before the land in ranch and the warning will go away.

**Comment by davionxiang** [comment]
@larshesel could you show me how to filter them out? I want to do that temporarily before your next release, since the warning is really annoying.

**Comment by larshesel** [comment]
Sure. The warning comes from this line when `reset_socket_opts/2` is being called: https://github.com/erlio/vernemq/blob/master/apps/vmq_server/src/vmq_ranch_proxy_protocol.erl#L554

That function is being called in lines https://github.com/erlio/vernemq/blob/master/apps/vmq_server/src/vmq_ranch_proxy_protocol.erl#L213 and https://github.com/erlio/vernemq/blob/master/apps/vmq_server/src/vmq_ranch_proxy_protocol.erl#L220.

So I guess the easiest would be to filter them out in either at the top of the `accept/2` function or in the `reset_socket_opts/2`. Or maybe even better would be in line https://github.com/erlio/vernemq/blob/master/apps/vmq_server/src/vmq_ranch_proxy_protocol.erl#L190 as this is where the Opts list is created in the first place... 

I haven't done it because I haven't had the time to test and verify...

**Comment by larshesel** [comment]
Closed via #549
